app/assets/javascripts/controllers/cloud_subnet/cloud_subnet_form_controller.js

```diff
@@ -61,27 +53,15 @@ ManageIQ.angular.app.controller('cloudSubnetFormController', ['$http', '$scope',
 
   $scope.filterNetworkManagerChanged = function(id) {
     miqService.sparkleOn();
-
-    $http.get('/cloud_subnet/cloud_subnet_networks_by_ems/' + id)
-      .then(getCloudSubnetNetworksByEmsComplete)
-      .catch(miqService.handleFailure);
-
-    $http.get('/cloud_subnet/cloud_tenants_by_ems/' + id)
-      .then(getCloudTenantsByEmsComplete)
-      .catch(miqService.handleFailure);
-
+    if (id) {
+      API.get("/api/cloud_networks?expand=resources&attributes=name,ems_ref&filter[]=ems_id=" + id).then(function(data) {
```

Comment:
I agree that the `CollectionsApi` is pretty naive. But I lack the experience with using our REST API to be able to formulate the requirements for a replacement.

The "URL standard" variant seems too verbose for me. It does not look elegant ;-)

The hash variant seems ok to me but right, it would have to deal with all the variants.

Maybe we need to collect more usage examples to get a better idea about where to move.

As for not hardcoding URLs: I can see benefit in 2 areas:
1) connecting to the API on a different server
2) using the same code for multiple entities

For 1): I don't think it is relevant at this point, with OPS UI we are pretty stuck with having the API on the same host as the UI code. And that is not going to change soon. And if we needed this we would probably do in https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/assets/javascripts/miq_api.js and would not change the callers at all.

For 2) We can pass the URL when needed. When refactoring some code to be used in more places, we can extract the URL into a parameter but I see no reason for doing it from the start of everywhere.

@AparnaKarve : I may be missing something, could you argue the benefit of passing the URLs around?